### PR TITLE
Improve TransactionMarkerChannelManager

### PR DESCRIPTION
**Pull request by eolivelli**
This patch adds failure handling on TransactionMarkerChannelHandler.
In case of failure on the socket we must fail every pending request, otherwise the system will be stuck

### CODE REVIEW

**Comment by Demogorgon314** [comment]
Please ping me if this PR is ready to review : ) 

Nice catch BTW.

**Comment by eolivelli** [comment]
@Demogorgon314 all tests passed.
I believe that this is a good first cleanup.

I figured out that most of the problems I saw happen when you have two brokers and the load balancer starts to unload topics and move them to another broker or when the bootstrap of one of the brokers is slower.
This is way I cannot reproduce my failures in simple unit tests.

I have not added tests yet, probably we should add tests with more then one broker and simulate failures, but it will take much time to code them.

I am continuing to run my system tests and see if this patch is enough to reach perfect stability (100% tests passing for a good number of times)

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionMarkerChannelManager.java [hunk 1]
```diff
@@ -235,13 +258,20 @@ public void addTxnMarkersToBrokerQueue(String transactionalId,
             CompletableFuture<Void> addFuture = new CompletableFuture<>();
             addressFutureList.add(addFuture);
             addressFuture.whenComplete((address, throwable) -> {
-                if (throwable != null) {
+                if (throwable != null)  {
```
Comment (Demogorgon314) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionMarkerChannelManager.java:261]:
```suggestion
                if (throwable != null) {
```

**Review by Demogorgon314 — APPROVED** [review]
Overall LGTM. Maybe we could add the units test in the next PR.

**Comment by eolivelli** [comment]
@Demogorgon314 I have committed your suggestion

**Pull request merged**